resources/rule.rb

```diff
@@ -17,40 +17,62 @@
 
 property :rule_name, String, name_property: true
 property :description, String, default: 'Firewall rule'
-property :localip, String
-property :localport, String
-property :remoteip, String
-property :remoteport, String
-property :dir, Symbol, default: :in, equal_to: [:in, :out]
+property :local_address, String
+property :local_port, String
+property :remote_address, String
+property :remote_port, String
+property :direction, Symbol, default: :inbound, equal_to: [:inbound, :outbound],
+          coerce: proc { |d| d.is_a?(String) ? d.downcase.to_sym : d }
 property :protocol, String, default: 'TCP'
-property :firewall_action, Symbol, default: :allow, equal_to: [:allow, :block, :bypass]
-property :profile, Symbol, default: :any, equal_to: [:public, :private, :domain, :any]
+property :firewall_action, Symbol, default: :allow, equal_to: [:allow, :block, :notconfigured],
+          coerce: proc { |f| f.is_a?(String) ? f.downcase.to_sym : f }
+property :profile, Symbol, default: :any, equal_to: [:public, :private, :domain, :any, :notapplicable],
+          coerce: proc { |p| p.is_a?(String) ? p.downcase.to_sym : p }
 property :program, String
 property :service, String
-property :interfacetype, Symbol, default: :any, equal_to: [:any, :wireless, :lan, :ras]
+property :interface_type, Symbol, default: :any, equal_to: [:any, :wireless, :wired, :remoteaccess],
+          coerce: proc { |i| i.is_a?(String) ? i.downcase.to_sym : i }
 
 load_current_value do
-  current_rule = shell_out("chcp 437 >null 2>&1 && cmd.exe /c netsh advfirewall firewall show rule name=\"#{rule_name}\"")
-
-  current_value_does_not_exist! if current_rule.stdout.strip == 'No rules match the specified criteria.'
+  load_state_cmd = load_firewall_state(rule_name)
+  output = powershell_out(load_state_cmd)
+  if output.stdout.empty?
+    current_value_does_not_exist!
+  else
+    state = Chef::JSONCompat.from_json(output.stdout)
```

Comment:
You should probably require the json compat module. It's coming in due to other things requiring it, but that might break some day

Reply:
fixed